apps/api/migrations/expire-at/expire-at.migration.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  MessageRepository,
+  NotificationRepository,
+  ExecutionDetailsRepository,
+  JobRepository,
+  OrganizationRepository,
+  EnvironmentRepository,
+  JobStatusEnum,
+} from '@novu/dal';
+import { addMonths } from 'date-fns';
+
+const messageRepository = new MessageRepository();
+const notificationRepository = new NotificationRepository();
+const jobRepository = new JobRepository();
+const executionDetailsRepository = new ExecutionDetailsRepository();
+const organizationRepository = new OrganizationRepository();
+const environmentRepository = new EnvironmentRepository();
+const now = Date.now();
+const expireAtOneMonth = addMonths(now, 1);
+const expireAtSixMonths = addMonths(now, 6);
+
+export async function createExpireAt() {
+  // eslint-disable-next-line no-console
+  console.log('start migration - add expireAt field');
+
+  // eslint-disable-next-line no-console
+  console.log('get organizations and its environments');
+
+  const organizations = await organizationRepository.find({});
+  for (const organization of organizations) {
+    const environments = await environmentRepository.findOrganizationEnvironments(organization._id);
+    for (const environment of environments) {
+      const query = {
+        _organizationId: organization._id,
+        _environmentId: environment._id,
+        expireAt: { $exists: false },
+      };
+      await messagesSetExpireAt(query);
+      await notificationExpireAt(query);
+    }
+  }
+
+  // eslint-disable-next-line no-console
+  console.log('end migration');
+}
+
+export async function messagesSetExpireAt(query) {
+  await messageRepository.update(
+    {
+      ...query,
+      channel: { $ne: 'in_app' },
+    },
+    { $set: { expireAt: expireAtOneMonth } }
+  );
+
+  await messageRepository.update(
+    {
+      ...query,
+      channel: 'in_app',
+    },
+    { $set: { expireAt: expireAtSixMonths } }
+  );
+}
+
+export async function notificationExpireAt(query) {
+  const excludedIds = await getExcludedNotificationIds(query);
```

Comment:
@ainouzgali I'm a bit concerned with this query when a customer has millions of events, won't this array will be huge? Is there other wise we can make the query without this find clause here?

Reply:
It will only include pending jobs (that are not digest related) - for example, a delay that did not finish yet. It shouldn't be big unless I'm missing something @scopsy

Reply:
Maybe we could use findBatch() here?

Reply:
Not here exactly but the idea in general in the flow.